components/eamxx/src/share/field/field.cpp

```diff
@@ -115,6 +115,46 @@ subfield (const int idim, const int index, const bool dynamic) const {
   return subfield(m_header->get_identifier().name(),idim,index,dynamic);
 }
 
+// slice at index idim, entries \in [index_beg, index_end]
```

Comment:
I would make end be the index _past_ the last slice. This would reflect what happens with the `end()` iterator from std containers and algorithms. Many devs would see "end" and immediately think that the required slices are `[beg,end)`. Also, it is more common to know the 1st index and the count, so that one would like to do `subfield(..., first, first+N)`, rather than add `-1` in the formula.

Reply:
Good call! My brain always defaults to math notation, but not the best choice here...